### PR TITLE
Update laser calibration tags for run2

### PR DESCRIPTION
**Pull request by mvicenzi**
This PR updates the tags of the timing calibration databases (`v2r0` -> `v2r1`) to pick up the latest PMT laser corrections for Run 2. This is also necessary for the extractions of the next-level corrections for Run 2, as cosmics corrections are computed on top of the laser ones.

**Note**: This update depends on new `pmt_cable_delays_data.db` and `pmt_laser_timing_data.db` files, which should be packaged in a concurrent `icarus_data` release. These files have been produced by @jedori0228 and tested by both of us.

Asking reviews from people that are (willingly or not) involved in this business.

### CODE REVIEW

**Comment by mvicenzi** [comment]
The laset timing tables relared to this PR are now available in `icarus_data` v09_79_00. Thanks to @jedori0228 and @SFBayLaser .

**Review by jedori0228 — APPROVED** [review]
Looks good to me, we will soon have the corresponding tag for `icarus_data` (v09_79_00).

**Review by SFBayLaser — APPROVED** [review]
The two db files needed for this release are in icarus_data v09_79_00 which will get build and distributed with the next keep p release. As I read this PR, this goes to the develop branch so will get picked up.

File: icaruscode/Timing/PMTTimingCorrectionsProvider.cxx
Comment (PetrilloAtWork) [inline icaruscode/Timing/PMTTimingCorrectionsProvider.cxx:222]: I assume this change is to avoid the header of messagefacility on each single line. Two ways to proceed here.
You may want to create a "permanent" instance of the `mf::LogInfo` object:
```cpp
      mf::LogInfo log(fLogCategory);
      log << "Dump information from database "
        << "\nchannel, trigger cable delay, reset cable delay, laser corrections, muons corrections";
      for( auto const & [key, value] : fDatabaseTimingCorrections ){
         log << "\n" << key << " "
            << ", " << value.triggerCableDelay
            << ", " << value.resetCableDelay
            << ", " << value.laserCableDelay
            << ", " << value.cosmicsCorrections;
      }
```
This will enclose the whole message in a single header.
Or, you may want to just replace the `mf::LogInfo` with `mf::Verbatim` in the loop, or in both the look and the intro line, depending on the effect you want. Each of the [messagefacility streams](https://github.com/art-framework-suite/messagefacility/blob/23731098651df6a6087a2d4d697b4baadba5590d/messagefacility/MessageLogger/MessageLogger.h#L207-L219) comes with a version with the "decoration" and a version without it; `mf::LogVerbatim` is the twin of `mf::LogInfo`.

File: icaruscode/Timing/timing_corrections_tags.fcl
Comment (PetrilloAtWork) [inline icaruscode/Timing/timing_corrections_tags.fcl:34]: This implies that we are applying on Run2 inconsistent laser and cosmic ray corrections.
Ideally we would have a null cosmic-ray-based correction for `run>=9301`.

**Review by PetrilloAtWork — COMMENTED** [review]
I left a technical comment and a more important one.
While I would appreciate a statement on the latter, I believe that this is ultimately a known behaviour, and on that assumption I am not blocking the merge.
I believe that in principle every new laser-based correction update should come with a null cosmic-ray-based one for the same period, until the latter is reextracted.

File: icaruscode/Timing/PMTTimingCorrectionsProvider.cxx
Comment (mvicenzi) [reply icaruscode/Timing/PMTTimingCorrectionsProvider.cxx:222]: Yes, that's precisely what I was trying to achieve to easily read the printed table for better debugging. I'll follow your suggestion and use one of those.

File: icaruscode/Timing/timing_corrections_tags.fcl
Comment (mvicenzi) [reply icaruscode/Timing/timing_corrections_tags.fcl:34]: Yes, it would be better. It would also simplify my work in extracting the cosmics corrections for Run 2 by not having to subtract already applied cosmics corrections (or having to ask to specifically produce calibration ntuples with cosmics corrections disabled). I'll ask @jedori0228 how painful it would be to add it...

**Comment by mvicenzi** [comment]
Following @PetrilloAtWork's comment, we decided to add a null table to avoid cosmics corrections meant for Run 1 being applied to Run 2. This updated file will need to make its way into `icarus_data` before merging this PR.

**Comment by mvicenzi** [comment]
Imported several commits from the code review of #628 , in order to match the code between production and develop.

**Comment by jedori0228** [comment]
`icarus_data` with tag `v09_79_01` now contains null table for Run>=9301 (table tag `v2r0`) for the cosmic timing table.

**Comment by mmrosenberg** [comment]
icaruscode's icarus_data dependency has now been updated to v09_79_01

**Comment by mmrosenberg** [comment]
trigger build

**Review by PetrilloAtWork — APPROVED** [review]
I am happy.

**Pull request merged**
